src/main/java/org/jabref/logic/importer/QueryConverter.java

```diff
@@ -0,0 +1,48 @@
+package org.jabref.logic.importer;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.jabref.logic.importer.fetcher.ComplexSearchQuery;
+
+import org.apache.lucene.index.Term;
+import org.apache.lucene.queryparser.flexible.core.QueryNodeException;
+import org.apache.lucene.queryparser.flexible.standard.StandardQueryParser;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.QueryVisitor;
+
+/**
+ * This class converts a query string written in lucene syntax into a complex search query.
+ *
+ * For simplicity this is limited to fielded data and the boolean AND operator.
+ */
+public class QueryConverter {
+
+    /**
+     * Converts the given query string into a complex query using lucene.
+     * Note: For unique fields, the alphabetically first instance in the query string is used in the complex query.
+     *
+     * @param queryString The given query string
+     * @return A complex query containing all fields of the query string
+     * @throws QueryNodeException Error during parsing
+     */
+    public ComplexSearchQuery convertQueryStringIntoComplexQuery(String queryString) throws QueryNodeException {
```

Comment:
Is the `QueryNodeException` helpful in diagnosing where the syntax error is, i.e. can it be converted to a meaningful error message shown to the user? If not, I would propose to catch it in the main body, change the return value to `Optional<Query>` and return an empty optional in case of a syntax error.